code/modules/mob/living/carbon/human/species/shadekin/shadekin_abilities.dm

```diff
@@ -33,6 +33,26 @@
 
 	var/ability_cost = 100
 
+//CHOMPADDITION Conditional shifting ased on observers and darkness
+	var/darkness = 1
+	var/turf/T = get_turf(src)
+	if(!T)
+		to_chat(src,"<span class='warning'>You can't use that here!</span>")
+		return FALSE
+
+	var/brightness = T.get_lumcount() //Brightness in 0.0 to 1.0
+	darkness = 1-brightness //Invert
+
+	var/watcher =-1
+	for(var/mob/living/carbon/human/watchers in view_or_range(7,get_turf(src),"range" ))
```

Comment:
This does not seem to be at all what's advertised. It does not care about whether the mob in question can actually see the shadekin, it simply grabs all mobs within the 7 tile range, even ones behind walls or ones incapable of otherwise seeing the kin. That and if I'm reading the code for range() correctly, then it absolutely does include the contents of the 'src'. Ideally, the solution would be to instead use viewers() centered on the kin with same range, and manually exclude any bellied people (ideally, ones bellied by other observers too, imo).

Reply:
Also, it should ideally at least check for whether 'watcher' actually has a player mind or not. A shadekin getting stuck in a monkey pen somewhere would be as hilarious as it would be disappointing.

Reply:
> This does not seem to be at all what's advertised. It does not care about whether the mob in question can actually see the shadekin, it simply grabs all mobs within the 7 tile range, even ones behind walls or ones incapable of otherwise seeing the kin. That and if I'm reading the code for range() correctly, then it absolutely does include the contents of the 'src'. Ideally, the solution would be to instead use viewers() centered on the kin with same range, and manually exclude any bellied people (ideally, ones bellied by other observers too, imo).

I don't know how to do such and encourage you to make a PR to my branch to add this change to the watcher system.

Reply:
Try `for(var/mob/living/carbon/human/watcher in viewers(7, src))`

Reply:
Try using `for(var/mob/living/carbon/human/watches in viewers(7, src))`